Source/player.cpp

```diff
@@ -490,31 +490,31 @@ void InitPlrGFXMem(int pnum)
 	const HeroClass c = player._pClass;
 
 	// STAND (ST: TOWN, AS: DUNGEON)
-	player._pNData = DiabloAllocPtr(std::max(GetPlrGFXSize(c, "ST"), GetPlrGFXSize(c, "AS")));
+	player._pNData = new std::uint8_t[std::max(GetPlrGFXSize(c, "ST"), GetPlrGFXSize(c, "AS"))];
 
 	// WALK (WL: TOWN, AW: DUNGEON)
-	player._pWData = DiabloAllocPtr(std::max(GetPlrGFXSize(c, "WL"), GetPlrGFXSize(c, "AW")));
+	player._pWData = new std::uint8_t[std::max(GetPlrGFXSize(c, "WL"), GetPlrGFXSize(c, "AW"))];
 
 	// ATTACK
-	player._pAData = DiabloAllocPtr(GetPlrGFXSize(c, "AT"));
+	player._pAData = new std::uint8_t[GetPlrGFXSize(c, "AT")];
 
 	// HIT
-	player._pHData = DiabloAllocPtr(GetPlrGFXSize(c, "HT"));
+	player._pHData = new std::uint8_t[GetPlrGFXSize(c, "HT")];
 
 	// LIGHTNING
-	player._pLData = DiabloAllocPtr(GetPlrGFXSize(c, "LM"));
+	player._pLData = new std::uint8_t[GetPlrGFXSize(c, "LM")];
 
 	// FIRE
-	player._pFData = DiabloAllocPtr(GetPlrGFXSize(c, "FM"));
+	player._pFData = new std::uint8_t[GetPlrGFXSize(c, "FM")];
 
 	// MAGIC
-	player._pTData = DiabloAllocPtr(GetPlrGFXSize(c, "QM"));
+	player._pTData = new std::uint8_t[GetPlrGFXSize(c, "QM")];
 
 	// DEATH
-	player._pDData = DiabloAllocPtr(GetPlrGFXSize(c, "DT"));
+	player._pDData = new std::uint8_t[GetPlrGFXSize(c, "DT")];
 
 	// BLOCK
-	player._pBData = DiabloAllocPtr(GetPlrGFXSize(c, "BL"));
+	player._pBData = new std::uint8_t[GetPlrGFXSize(c, "BL")];

```

Comment:
I'd be partial to using std::unique_ptr. If/when this gets refactored into classes, we can use RAII to free them.

Reply:
This struct is being memset to 0 so we can't use non-POD types in it yet

Reply:
You can do in-place construction after memset.

Reply:
memset on non-POD types is UB

Reply:
You can use a a separate variable for the memset, then do in-place construction and return the actual struct.

Reply:
Although if it were me, it just initialize every one of those ints to 0.

Reply:
Further, the destructor would free the pointers for you.

Reply:
I'd definitely prefer just adding member initializers to each field and then doing `plr = {}` instead of memset. That memset is not a hotspot and does not require the placement new dance.

This came up in #1795 as well; wanna take this one?

Reply:
I'm very busy this week, but if nobody does it the mean time, I'll do it.